src/include/stir/LORCoordinates.inl

```diff
@@ -346,8 +346,19 @@ find_LOR_intersections_with_cylinder(LORAs2Points<coordT1>& intersection_coords,
   auto l2 = static_cast<coordT2>((-b - root) / a);
   
   // TODO won't work when coordT1!=coordT2
-  intersection_coords.p1() = d*l1 + c1;
-  intersection_coords.p2() = d*l2 + c1;
+  const CartesianCoordinate3D<coordT2> distance1 = d * l1 + c1 - coords.p1();
```

Comment:
TODO: check if this can be done just with math to improve performance.

Reply:
seems to be me that due to the parametrisation `c = l*d+c1`, the smallest `l` is going to be closest to `c1`. Therefore, chosing the negative root as `l1` should do the trick, i.e. the old code *always* swapped.

Reply:
This makes sense and matches what I've seen when debugging through it. I'll test with just swapping l1 and l2.